chain/rosetta-rpc/src/adapters/nep141.rs

```diff
@@ -0,0 +1,209 @@
+                        let custom = crate::models::FtEvent {
+                            affected_id: AccountIdentifier::from_str(&transfer_event.new_owner_id)?,
+                            involved_id: Some(AccountIdentifier::from_str(
+                                &transfer_event.old_owner_id,
+                            )?),
+                            delta: transfer_event.amount.parse::<i64>()?,
+                            cause: "TRANSFER".to_string(),
+                            memo: transfer_event
+                                .memo
+                                .as_ref()
+                                .map(|s| s.escape_default().to_string()),
+                            symbol: currency.symbol.clone(),
+                            decimals: currency.decimals,
+                        };
+                        ft_events.push(build_event(base, custom).await?);
+                    }
+                }
+            }
+        }
+    }
+    Ok(ft_events)
+}
+
+pub(crate) async fn get_fungible_token_balance_for_account(
+    view_client_addr: &actix::Addr<near_client::ViewClientActor>,
+    block_header: &near_primitives::views::BlockHeaderView,
+    contract_address: &String,
+    account_identifier: &AccountIdentifier,
+) -> crate::errors::Result<u128> {
+    let method_name = "ft_balance_of".to_string();
+    let account_id_for_args = account_identifier.clone().address.to_string();
+    let args = serde_json::json!({
+        "account_id":account_id_for_args,
+    })
+    .to_string()
+    .into_bytes();
+    let block_reference =
+        near_primitives::types::BlockReference::BlockId(BlockId::Hash(block_header.hash));
+    let request = near_primitives::views::QueryRequest::CallFunction {
+        account_id: near_account_id::AccountId::from_str(contract_address)?,
+        method_name,
+        args: near_primitives::types::FunctionArgs::from(args),
+    };
+    let query_response = view_client_addr
+        .send(near_client::Query { block_reference, request }.with_span_context())
+        .await?
+        .map_err(|e| crate::errors::ErrorKind::InternalInvariantError(e.to_string()))?;
+    let call_result = if let near_primitives::views::QueryResponseKind::CallResult(result) =
+        query_response.kind
+    {
+        result.result
+    } else {
+        return Err(crate::errors::ErrorKind::InternalInvariantError(format!(
+            "Couldn't retrieve ft_balance of {:?} on address {:?}",
+            account_identifier.address.clone(),
+            contract_address.clone(),
+        )));
+    };
+    let serde_call_result = serde_json::from_slice(&call_result).or(Err(
+        crate::errors::ErrorKind::InternalInvariantError(format!(
+            "Couldn't read the value from the contract {:?}, for the account {:?}",
+            contract_address.clone(),
+            account_identifier.address.clone(),
+        )),
+    ))?;
```

Comment:
(optional)
You are dropping a very useful [serde_json Error](https://docs.rs/serde_json/latest/serde_json/struct.Error.html) here. Consider displaying it in the error message to help debugging any potential errors. For example when a custom FT contract prints invalid JSON events.

Ideal would be to add a parsing error in `ErrorKind` and add the serde errer as a `#[source]`.

But even just including it in the string of `InternalInvariantError` could be a benefit IMO.

But again - optional.